### PR TITLE
Add drmemtrace trace entry docs

### PR DESCRIPTION
**Pull request by derekbruening**
Adds explicit documentation for the drmemtrace trace entry record
types.  This is a common topic asked about by new users of our tracing
framework.

### CODE REVIEW

File: clients/drcachesim/drcachesim.dox.in [hunk 1]
```diff
@@ -81,6 +82,86 @@ analysis tools (see \ref sec_drcachesim_newtool).
...
+trace.  Each marker record contains two additional fields:
+
+- \p marker_type - identifies the type of marker
+- \p marker_value - carries the value of the marker
+
+Some of the more important markers are:
+
+- #TRACE_MARKER_TYPE_KERNEL_EVENT - This identifies kernel-initiated control transfers such as signal delivery.  The next instruction record is the start of the handler for a kernel-initiated event.  The value of this type of marker contains the program counter at the kernel interruption point.  If the interruption point is just after a branch, this value is the target of that branch.
+
+- #TRACE_MARKER_TYPE_KERNEL_XFER - This identifies a system call that changes control flow, such as a signal return..
```
Comment (abhinav92003) [inline clients/drcachesim/drcachesim.dox.in:156]: nit: multiple '.'s.

File: clients/drcachesim/drcachesim.dox.in [hunk 2]
```diff
@@ -81,6 +82,86 @@ analysis tools (see \ref sec_drcachesim_newtool).
...
+
+- #TRACE_MARKER_TYPE_TIMESTAMP - The marker value provides a timestamp for this point of the trace (in units of microseconds since Jan 1, 1601 UTC). This value can be used to synchronize records from different threads. It is used in the sequential analysis of a multi-threaded trace.
+
+- #TRACE_MARKER_TYPE_CPU_ID - The marker value contains the CPU identifier on which the subsequent records were collected. It is useful to help track thread migrations occurring during execution.
+
+- #TRACE_MARKER_TYPE_FUNC_ID, #TRACE_MARKER_TYPE_FUNC_RETADDR, #TRACE_MARKER_TYPE_FUNC_ARG, #TRACE_MARKER_TYPE_FUNC_RETVAL - These markers are used to capture information about function calls.  Which functions to capture must be explicitly selected at tracing time.  Typical candiates are heap allocation and freeing functions.  See sec_drcachesim_funcs.
```
Comment (abhinav92003) [inline clients/drcachesim/drcachesim.dox.in:162]: \ref sec_drcachesim_funcs`

File: clients/drcachesim/drcachesim.dox.in [hunk 3]
```diff
@@ -81,6 +82,86 @@ analysis tools (see \ref sec_drcachesim_newtool).
...
+
+
```
Comment (abhinav92003) [inline clients/drcachesim/drcachesim.dox.in:164]: Add link to the page that exhaustively describes all markers.

File: clients/drcachesim/drcachesim.dox.in (hunk 1 above)
Comment (derekbruening) [reply clients/drcachesim/drcachesim.dox.in:156]: Fixed.

File: clients/drcachesim/drcachesim.dox.in (hunk 2 above)
Comment (derekbruening) [reply clients/drcachesim/drcachesim.dox.in:162]: Done and confirmed it makes a proper link.

File: clients/drcachesim/drcachesim.dox.in (hunk 3 above)
Comment (derekbruening) [reply clients/drcachesim/drcachesim.dox.in:164]: Done.

**Review by abhinav92003 — APPROVED** [review]

**Comment by derekbruening** [comment]
Failure is known flake client.winxfer #4732

**Pull request merged**